.solargraph.yml

```diff
@@ -0,0 +1,22 @@
+---
+include:
+- "lib/**/*.rb"
+exclude:
+- spec/**/*
+- test/**/*
+- vendor/**/*
+- ".bundle/**/*"
+require: []
+domains: []
+reporters:
+- rubocop
+- require_not_found
+formatter:
+  rubocop:
+    cops: safe
+    except: []
+    only: []
+    extra_args: []
+require_paths: []
+plugins: []
+max_files: 8000
```

Comment:
This can be set to 0 to remove the file limit entirely. I did this for a while when I was using SolarGraph with VSCode before I migrated back over to RubyMine.

Most of the team uses RubyMine minus me before I switched back over so I'd probably be the best one to test this out since I already have a setup and detailed my config over at https://tekwizz123.hashnode.dev/my-vscode-workflow-for-ruby-programming-on-metasploit as the following:

```
---
include:
  - "**/*.rb"
exclude:
  - spec/**/*
  - test/**/*
  - vendor/**/*
  - ".bundle/**/*"
  - data/**/*
  - db/**/*
  - external/**/*
  - plugins/**/*
  - scripts/**/* # Some of this is old and may not need indexing???
require: []
domains: []
reporters:
  - rubocop
  - require_not_found
formatter:
  rubocop:
    cops: safe
    except: []
    only: []
    extra_args: []
require_paths: []
plugins: []
max_files: 0
```

Don't know if perhaps we want to combine some of your suggestions with some of the edits in there? For instance I think not indexing things like `scripts/`

Reply:
I agree with a lot of these exclusions, I'm uncertain about reading all ruby files, especially the modules since there are so many. If the rest of the team uses rubymine, configuration for rubymine would be appreciated

Reply:
RubyMine can't use LSPs at the moment. This is a conscious decision on their part from what I can tell. Whilst there is an LSP plugin, I haven't tried it in part cause that plugin isn't maintained.